클라이언트 배포 서버 구축 및 자동배포 워크플로우 추가

Park-SooYeon opened the pull request:
# 클라이언트 배포 서버 구축 및 자동배포 워크플로우 추가

## 📎 해당 이슈

이슈 링크 (#316)

## 🛠 구현 내용 

- 클라이언트 배포 서버 구축
- 클라이언트 자동배포 워크플로우 추가

## 🙋‍ 리뷰어 참고 사항

Park-SooYeon committed:
[FEAT] 클라이언트 자동배포 워크플로우 추가

youngxpepp commented on .github/workflows/client-server.yml:
```diff
@@ -0,0 +1,39 @@
+    working-directory: ./client
+jobs:
+  deploy:
+    if: ${{ github.event_name == 'push' }}
```
job에 deploy만 있어서 조건문은 필요 없을 듯 해요~

Park-SooYeon replied:
commit 1f1671d
수정하였습니다!

youngxpepp commented on .github/workflows/client-server.yml:
```diff
@@ -0,0 +1,39 @@
+        uses: actions/setup-node@v2.1.2
+        with:
+          node-version: 12.x
+      - run: npm install --only prop
```
이건 어떤 옵션인가요?

Park-SooYeon replied:
dependencies만 설치하도록 한 옵션이였는데 잘못됬군요ㅠ 수정하겠습니다!

youngxpepp reviewed (changes requested):
포크 떠서 배포 과정 테스트 해보셨을까요?
마지막에 scp로 파일 전송한 다음에 nginx에 대한 로직이 없는 것 같은데, nginx에서 파일 변경사항을 자동으로 파악하나요?

Park-SooYeon commented:
build된 파일의 위치가 변경되지 않는 이상, nginx 설정은 push할때마다 바뀔 이유가 없을 것이라 생각해서 추가하지 않았습니다.
포크떠서 배포과정 테스트 해보겠습니다.!

Park-SooYeon committed:
[FIX] 자동배포 테스트 후, workflow 수정

- webpack 파일의 쓰기 권한이 없어 npm install 후, overwrite를 하게되면 권한 오류가 발생
- 파일을 옮긴 후, npm install, build 수행하도록 수정

Park-SooYeon committed:
[FIX] 필요없는 workflow 조건 제거

youngxpepp commented on .github/workflows/client-server.yml:
```diff
@@ -0,0 +1,43 @@
+          port: ${{ secrets.CLIENT_SERVER_PORT }}
+          username: ${{ secrets.CLIENT_SERVER_USERNAME }}
+          password: ${{ secrets.CLIENT_SERVER_PASSWORD }}
+          script: |
+            export NVM_DIR=~/.nvm
+            source ~/.nvm/nvm.sh
+            cd ~/client
+            npm install --production
+            npm run build
```
배포 서버 자원을 쓰는 것보다 미리 빌드해서 건네주는게 낫지 않을까요?

Park-SooYeon replied:
webpack 모듈 파일 권한이 읽기만 가능해서 그런지 서버로 복사하는 과정에서 overwrite하는 도중에 권한 에러가 뜨더라구요..😢
그렇다고 파일 권한을 수정하는 건 또 아닌거 같아서 어쩔수 없이 서버에서 빌드해주는 것으로 했습니다..
이게 아니면 더 좋은 방법이 있을까요?👀

youngxpepp replied:
웹팩 모듈을 같이 보내야 하는 이유가 있을까요? 
Project04-B=Whale/client/build/**/* 해당 경로에 있는 파일만 보내주면 되지 않나용?

youngxpepp replied:
웹팩 모듈을 같이 보내야 하는 이유가 있을까요?
Project04-B=Whale/client/build/**/* 해당 경로에 있는 파일만 보내주면 되지 않나용?

Park-SooYeon replied:
흠..생각해보니 그러네요.🤔
애당초 빌드된 파일을 읽어서 보여주는거니 빌드된 파일만 서버에 보내줘도 될것 같네요!
수정하겠습니다!!

Park-SooYeon replied:
commit 1e68a63
빌드된 파일만 서버에 저장하도록 수정했습니다!

Park-SooYeon committed:
[FIX] build된 파일만 클라이언트 서버에 저장하도록 변경
